rust/Earthfile

```diff
@@ -80,10 +80,13 @@ INSTALL_CARGO_SWEEP:
 RUN_WITH_CACHE:
     COMMAND
     ARG command
-    RUN --mount=type=cache,mode=0777,id="earthly-cargo-home-registry",sharing=shared,target=$CARGO_HOME/registry \
-        --mount=type=cache,mode=0777,id="earthly-cargo-home-git",sharing=shared,target=$CARGO_HOME/git \
-        --mount=type=cache,mode=0777,id="earthly-cargo-deps",sharing=shared,target=/tmp/earthly \
+    ARG ORIGINAL_CARGO_HOME=$CARGO_HOME
+    ENV CARGO_HOME="/earthly/.cargo"
```

Comment:
What is the reason for relocating cargo home? Does the approach not work just as well with the default `$CARGO_HOME`?

Also, I hadn't appreciated before that this cache would be shared between all jobs running on the same host, so potentially completely unrelated workflows or even projects (if they shared). There's a danger that one project may accidentally rely on tools installed to the cache by another project instead of installing themselves and break if the cache is cleared. Unfortunate but unavoidable given the caching model perhaps?

Reply:
> We need $CARGO_HOME/.package-cache to be inside the shared cache and we cannot mount individual files, so needed to mount a parent folder

Sure, but why is `/earthly/.cargo` a better choice than the default `$HOME/.cargo`?

> As I see it, it's the same as working locally: 

I would generally expect a CI environment to be more isolated and deterministic than my workstation environment. Sharing a cache between workflows on a single project seems like an acceptable performance trade off, but sharing a cache between multiple different projects seems like a potential foot gun to me. Other CI systems tend to allow you to include the repo name or even the branch name[0^] into the cache key.

[0^]: Sometimes they allow you to fork from a parent branch's cache as a starting point for a new branch.

Reply:
makes sense

Reply:
>Sure, but why is /earthly/.cargo a better choice than the default $HOME/.cargo?

`$HOME/.cargo` is present in the target filesystem. If we create a mount cache on top of it, we lose the cargo binaries